deepspeed/runtime/utils.py

```diff
@@ -181,12 +181,13 @@ def get_norm_with_moe_layers_fast(all_groups_norm, group):
 class CheckOverflow(object):
     '''Checks for overflow in gradient across parallel process'''
 
-    def __init__(self, param_groups=None, mpu=None, zero_reduce_scatter=False, deepspeed=None):
+    def __init__(self, param_groups=None, mpu=None, zero_reduce_scatter=False, deepspeed=None, partition_grads=False):
```

Comment:
Passing deepseed engine into a submodule is not a good design and create all sorts of cyclic reference issues. It is better to pass the specific attributes that are needed, such as `enable_backward_allreduce`